test/plugin/test_exception_detector.rb

```diff
@@ -350,4 +350,46 @@ def test_json_messages
       assert_equal(expected_json(s.actual_field, s.output), out, s.desc)
     end
   end
+
+  def test_max_lines_limit
+    max_lines = JAVA_EXC_PART1.lines.length
+    out = []
+    buffer = Fluent::TraceAccumulator.new(nil,
+                                          [:all],
+                                          max_lines: max_lines) do |_, m|
```

Comment:
Just realized: instead of relying on the default limit, you should specify "unlimited" for `max_bytes` here and `max_lines` below (especially true since you've now changed the default limit for `max_lines` from "unlimited").